### PR TITLE
feat(pass-plan): pre-pass component with state automation

### PR DESCRIPTION
**Pull request by FMorrison87**
This PR includes the automation for the pre-pass component and more fine-grained management of the pre-pass --> pass state swap. A useEffect handles swapping the list item rows into a 'completed state' based on timers.

### CODE REVIEW

**Review by NicDominguez — COMMENTED** [review]
I'm going to make some changes in my own commits to save some time.

**Review by NicDominguez — APPROVED** [review]

File: src/Command/Components/PassPlan/PassPlan.tsx [hunk 1]
```diff
@@ -28,62 +23,18 @@ const PassPlan = () => {
           <RuxOption label="Automatic" value="" />
         </RuxSelect>
       </div>
-      <div className="banner">Pre-Pass</div>
-      <div className="rux-tree-header">
-        <span>Step</span>
-        <span>Instruction</span>
+      <div>
+        <div className={`banner ${pass}`}>{pass}</div>
```
[the remark this answers is not in the document]
Comment (FMorrison87) [reply src/Command/Components/PassPlan/PassPlan.tsx:27]: It does change in the previous version of the app, and definitely up for debate. I added in that style more to call attention to the fact that we are changinf from pre-pass into pass.

File: src/Command/Components/PassPlan/PassList/PassList.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    <>
+          
+        <RuxTree>
+          <RuxTreeNode>
+            <div className="rux-tree-node_prefix" slot="prefix">
+              1
+            </div>
+            <div className="rux-tree-node_content">
+              <RuxCheckbox />
+              A nice hat
```
[the remark this answers is not in the document]
Comment (FMorrison87) [reply src/Command/Components/PassPlan/PassList/PassList.tsx]: It is! (This is removed in my next PR)

File: src/Command/Components/PassPlan/PrePassList/PrePassList.tsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    // if the current list item is the length of the state function array, set the pass from 'pre-pass' into 'pass'
+    if (currentListItem === stateFunctionArray.length) {
+      setPass("Pass");
+      return;
+    }
+
+    // every 7 milliseconds for each progress bar, set the progress value from 0-100 (filling the bar)
+    let value: number = 0;
+    setInterval(() => {
+      if (value === 101) return;
```
[the remark this answers is not in the document]
Comment (FMorrison87) [reply src/Command/Components/PassPlan/PrePassList/PrePassList.tsx]: All set in my next PR.

**Pull request merged**